common/common.go

```diff
@@ -610,22 +610,40 @@ func GetDefaultDataDir(env string) string {
 	return path.Join(os.ExpandEnv("$HOME"), ".gocryptotrader")
 }
 
-// CheckDir checks to see if a particular directory exists
-// and attempts to create it if desired, if it doesn't exist
-func CheckDir(dir string, create bool) error {
+// CreateDir creates a directory based on the supplied parameter
+func CreateDir(dir string) error {
 	_, err := os.Stat(dir)
 	if !os.IsNotExist(err) {
 		return nil
 	}
 
-	if !create {
-		return fmt.Errorf("directory %s does not exist. Err: %s", dir, err)
-	}
-
 	log.Warnf("Directory %s does not exist.. creating.", dir)
-	err = os.MkdirAll(dir, 0777)
+	return os.MkdirAll(dir, 0770)
+}
+
+// ListAllDir lists all the directories and files in an array
+func ListAllDir(directory string) ([]string, error) {
```

Comment:
You have called this function ListAllDir's but due to your code on line 628 it will only append folders

not handling the error here will also cause the problem to segfault if the folder doesn't exist